waku/v2/waku_relay/protocol.nim

```diff
@@ -62,51 +144,28 @@ proc initProtocolHandler(w: WakuRelay) =
   w.handler = handler
   w.codec = WakuRelayCodec
 
-method initPubSub(w: WakuRelay) {.raises: [InitializationError].} =
-  ## NOTE: This method overrides GossipSub initPubSub method; it called by the
-  ##  parent protocol, PubSub.
-  debug "init waku relay"
-
-  # After discussions with @sinkingsugar: This is essentially what is needed for
-  # the libp2p `StrictNoSign` policy
-  w.anonymize = true
-  w.verifySignature = false
-  w.sign = false
+proc new*(T: type WakuRelay, switch: Switch): WakuRelayResult[T] =
 
-  procCall GossipSub(w).initPubSub()
-
-  w.initProtocolHandler()
-
-
-proc new*(T: type WakuRelay, switch: Switch, triggerSelf: bool = true): WakuRelayResult[T] =
-
-  var wr: WakuRelay
+  var w: WakuRelay
   try:
-    wr = WakuRelay.init(
+    w = WakuRelay.init(
       switch = switch,
-      msgIdProvider = defaultMessageIdProvider,
-      triggerSelf = triggerSelf,
-      sign = false,
+      anonymize = true,
       verifySignature = false,
-      maxMessageSize = MaxWakuMessageSize
+      sign = false,
+      triggerSelf = true,
+      msgIdProvider = defaultMessageIdProvider,
+      maxMessageSize = MaxWakuMessageSize,
+      parameters = GossipsubParameters
     )
-  except InitializationError:
-    return err("initialization error: " & getCurrentExceptionMsg())
-
-    # TODO: Add a function to validate the WakuMessage integrity
-    # # Rejects messages that are not WakuMessage
-    # proc validator(topic: string, message: messages.Message): Future[ValidationResult] {.async.} =
-    #   let msg = WakuMessage.decode(message.data)
-    #   if msg.isOk():
-    #     return ValidationResult.Accept
-    #   return ValidationResult.Reject
 
-    # # Add validator to all default pubsub topics
-    # for pubSubTopic in defaultPubsubTopics:
-    #   wr.addValidator(pubSubTopic, validator)
+    procCall GossipSub(w).initPubSub()
+    w.initProtocolHandler()
 
-  ok(wr)
+  except InitializationError:
+    return err("initialization error: " & getCurrentExceptionMsg())
 
+  ok(w)
```

Comment:
Not a big deal at all but for I personally prefer being explicit and always use `return`. Just for new incomers to Nim :)
```suggestion
  return ok(w)
```

Reply:
yep, i actually prefer to be explicit with this
dc70df97f859a4fa907db0be13bfd56d7a23dce8